db/compaction/compaction_picker_universal.cc

```diff
@@ -841,34 +841,47 @@ Compaction* UniversalCompactionBuilder::PickDeleteTriggeredCompaction() {
   std::vector<CompactionInputFiles> inputs;
 
   if (vstorage_->num_levels() == 1) {
-#if defined(ENABLE_SINGLE_LEVEL_DTC)
     // This is single level universal. Since we're basically trying to reclaim
     // space by processing files marked for compaction due to high tombstone
     // density, let's do the same thing as compaction to reduce size amp which
     // has the same goals.
-    bool compact = false;
+    int start_index = -1;
 
     start_level_inputs.level = 0;
     start_level_inputs.files.clear();
     output_level = 0;
-    for (FileMetaData* f : vstorage_->LevelFiles(0)) {
-      if (f->marked_for_compaction) {
-        compact = true;
+    // Find the first file marked for compaction. Ignore the last file
+    for (size_t loop = 0; loop + 1 < sorted_runs_.size(); loop++) {
+      SortedRun* sr = &sorted_runs_[loop];
+      if (sr->being_compacted) {
+        continue;
       }
-      if (compact) {
+      FileMetaData* f = vstorage_->LevelFiles(0)[loop];
```

Comment:
I am confused by this code.  Looking at how sorted_runs_ is built, it appears that the value for loop could be greater than the depth of LevelFiles[0].  Won't this run off the end of the LevelFiles(0)[] array?